Kommunicate/Classes/KMConversationListViewController.swift

```diff
@@ -0,0 +1,642 @@
+        message.groupId = groupId
+        let info = notification.userInfo
+        let alertValue = info?["alertValue"]
+        guard let updateUI = info?["updateUI"] as? Int else { return }
+        if updateUI == Int(APP_STATE_ACTIVE.rawValue), self.isViewLoaded, self.view.window != nil {
+            guard let alert = alertValue as? String else { return }
+            let alertComponents = alert.components(separatedBy: ":")
+            if alertComponents.count > 1 {
+                message.message = alertComponents[1]
+            } else {
+                message.message = alertComponents.first
+            }
+            self.viewModel.addMessages(messages: [message])
+        } else if updateUI == Int(APP_STATE_INACTIVE.rawValue) {
+            // Coming from background
+
+            guard groupId != nil || conversationId != nil else { return }
+            self.launchChat(groupId: groupId)
+        }
+    }
+
+    @objc func updateMessageList(notification: NSNotification) {
+        print("Reloadtable notification received")
+        guard let list = notification.object as? [Any] else { return }
+        self.viewModel.updateMessageList(messages: list)
+    }
+
+    @objc func updateUserDetails(notification: NSNotification) {
+        print("update user detail notification received")
+        guard let userId = notification.object as? String else { return }
+        print("update user detail")
+        self.viewModel.updateUserDetail(userId: userId) { (success) in
+            if success {
+                self.tableView.reloadData()
+            }
+        }
+    }
+
+    @objc func updateChannelName(notification: NSNotification) {
+        print("update group name notification received")
+        guard self.view.window != nil else { return }
+        print("update group detail")
+        self.tableView.reloadData()
+    }
+
+    public override func removeObserver() {
+        if alMqttConversationService != nil {
+            alMqttConversationService.unsubscribeToConversation()
+        }
+    }
+
+    func setupBackButton() {
+        let back = LocalizedText.leftBarBackButtonText
+
+        let leftBarButtonItem = UIBarButtonItem(title: back, style: .plain, target: self, action: #selector(customBackAction))
+
+        if !configuration.hideBackButtonInConversationList {
+            navigationItem.leftBarButtonItem = leftBarButtonItem
+        }
+    }
+
+    func setupNavigationRightButtons() {
+        let navigationItems = configuration.navigationItemsForConversationList
+
+        var rightBarButtonItems: [UIBarButtonItem] = []
+        if configuration.isMessageSearchEnabled {
+            let barButton = UIBarButtonItem(
+                image: UIImage(named: "search", in: Bundle.kommunicate, compatibleWith: nil),
+                style: .plain,
+                target: self, action: #selector(searchTapped)
+            )
+            rightBarButtonItems.append(barButton)
+        }
```

Comment:
Can you directly add a new chat button here instead of passing it through `navigationItemsForConversationList`? This way we won't be dependent on `defaultConfiguration` for new chat button, and  we can use `hideStartChatButton` config here just like `isMessageSearchEnabled`.